src/python/pants/core/util_rules/strip_source_roots.py

```diff
@@ -93,20 +96,26 @@ def determine_source_root(path: str) -> str:
         return PurePath(path).parent.as_posix()
 
     if request.representative_path is not None:
-        resulting_digest = await Get[Digest](
-            RemovePrefix(
-                request.snapshot.digest, determine_source_root(request.representative_path),
-            )
-        )
-        resulting_snapshot = await Get[Snapshot](Digest, resulting_digest)
-        return SourceRootStrippedSources(snapshot=resulting_snapshot)
+        source_root = determine_source_root(request.representative_path)
+        is_repo_level_source_root = source_root == ""
```

Comment:
I think you can safely inline this into the `if` condition. It's more intuitive than having to parse this long variable name (what does "repo-level" mean? the reader may be asking themselves).